batch/batch/front_end/query.py

```diff
@@ -0,0 +1,591 @@
+import abc
+from typing import Any, List, Optional, Tuple, Type
+
+from hailtop.utils import parse_timestamp_msecs
+
+from ..exceptions import QueryError
+from .operators import (
+    ComparisonOperator,
+    GreaterThanEqualOperator,
+    GreaterThanOperator,
+    LessThanEqualOperator,
+    LessThanOperator,
+    MatchOperator,
+    Operator,
+    PartialMatchOperator,
+    parse_operator,
+)
+
+CURRENT_QUERY_VERSION = 1
+
+
+state_search_term_to_states = {
```

Comment:
Let's use an `Enum` instead of strings for these states. Some of these strings like `Running` are written multiple times and I imagine that would be an easy footgun if one of them were misspelled.